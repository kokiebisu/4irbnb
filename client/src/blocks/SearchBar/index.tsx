import React from 'react';

// block
import SearchBarLarge from './searchbar.large';
import SearchBarSmall from './searchbar.small';

type SearchButtonProps = {
  type: string;
};

interface typeProps {
  [key: string]: JSX.Element;
}

export const SearchButton = ({ type, ...props }: SearchButtonProps) => {
  const types: typeProps = {
    mobile: <SearchBarLarge {...props} />,
    normal: <SearchBarSmall {...props} />,
  };
  return types[type];
};
