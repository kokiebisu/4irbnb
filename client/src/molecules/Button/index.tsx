import React from 'react';

// atoms
import { Button } from 'atoms';

// blocks
import SearchButton from 'molecules/Button/button.search';
import CookieButton from 'molecules/Button/button.cookie';
import GlobeButton from 'molecules/Button/button.globe';

export type ButtonProps = {
  type?: string;
  onPress: () => void;
  sm?: boolean;
  md?: boolean;
  lg?: boolean;
  inverse?: boolean;
  name?: string;
};

interface mapProps {
  [key: string]: JSX.Element;
}

/**
 * Use a button when you want to perform specific actions
 **/
export const CustomButton = ({ type, ...props }: ButtonProps) => {
  if (!type) {
    return <Button {...props} />;
  }
  const types: mapProps = {
    search: <SearchButton {...props} />,
    cookie: <CookieButton {...props} />,
    globe: <GlobeButton {...props} />,
  };
  return types[type];
};
