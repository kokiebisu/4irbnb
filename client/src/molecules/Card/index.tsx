import React from 'react';
import styled, { css } from 'styled-components';
import theme from 'styled-theming';
import { darken } from 'polished';
import { Box, Text, Img } from 'atoms';
import { colors } from 'styles';

type CardProps = {
  image: string;
  title?: string;
  description?: string;
  subtitle?: string;
  size?: string;
};

const background = theme('mode', {
  light: colors.white,
  dark: darken(0.8, colors.gray),
});

const primary = theme('mode', {
  light: darken(0.8, colors.gray),
  dark: darken(0.4, colors.white),
});

const secondary = theme('mode', {
  light: colors.gray,
  dark: darken(0.4, colors.white),
});

const shadow = theme('mode', {
  light: 'rgba(0, 0, 0, 0.15)',
  dark: 'rgba(0, 0, 0, 0.15)',
});

const Wrapper = styled(Box)`
  height: 100%;
  background-color: ${background};
  border-radius: 16px;
  box-shadow: ${shadow} 0px 2px 8px;
`;

const ImageWrapper = styled(Box)`
  object-fit: cover;

  & > img {
    border-top-left-radius: 16px;
    border-top-right-radius: 16px;
  }
`;

const Content = styled(Box)`
  &.large {
    width: 50%;
    text-align: left;
    padding: 0 18px 12px 18px;
  }
  &.normal {
    padding: 0 18px 12px 18px;
  }
`;

const extend = {
  title: css`
    font-weight: 700;
    color: ${primary};
    padding: 4px 0 2px 0;
  `,
  subtitle: css`
    padding: 14px 0;
    color: ${primary};
    font-weight: 500;
    line-height: 1.3;
    font-size: 15px;
  `,
  description: css`
    padding-top: 2px;
    font-size: 13px;
    font-weight: 300;
    color: ${secondary};
    line-height: 1.4;
  `,
};

export default ({
  image,
  title,
  subtitle,
  description,
  size,
  ...props
}: CardProps) => {
  return (
    <Wrapper {...props}>
      <ImageWrapper>
        <Img src={image} alt='card-image' />
      </ImageWrapper>
      <Content className={size === 'lg' ? 'large' : 'normal'}>
        {title && <Text styles={extend.title}>{title}</Text>}
        {subtitle && <Text styles={extend.subtitle}>{subtitle}</Text>}
        {description && <Text styles={extend.description}>{description}</Text>}
      </Content>
    </Wrapper>
  );
};
