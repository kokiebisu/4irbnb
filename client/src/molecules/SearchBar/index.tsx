import React from 'react';

// block
import SearchBarLarge from 'molecules/SearchBar/searchbar.large';
import SearchBarMedium from 'molecules/SearchBar/searchbar.medium';
import SearchBarSmall from 'molecules/SearchBar/searchbar.small';

type SearchBarProps = {
  type: string;
  onPress: () => void;
};

interface mapProps {
  [key: string]: JSX.Element;
}

export const SearchBar = ({ type, ...props }: SearchBarProps) => {
  const types: mapProps = {
    sm: <SearchBarMedium {...props} />,
    md: <SearchBarSmall {...props} />,
    lg: <SearchBarLarge {...props} />,
  };
  return types[type];
};
