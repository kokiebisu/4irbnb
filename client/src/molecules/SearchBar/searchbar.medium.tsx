import React from 'react';
import styled, { css } from 'styled-components';
import theme from 'styled-theming';
import { darken } from 'polished';
import { colors } from 'styles';
import { Box, Button, Icon, Text } from 'atoms';

const primary = theme('mode', {
  light: colors.gray,
});

const secondary = theme('mode', {
  light: darken(0.7, colors.gray),
});

const Wrapper = styled(Button)`
  background-color: transparent;
  box-shadow: rgb(0, 0, 0, 0.15) 0px 5px 12px;
  height: 50px;
  width: 100%;
  padding: 0 10px 0 18px;
  border-radius: 24px;
`;

const Flex = styled(Box)`
  display: flex;
  align-items: center;
`;

const CustomIcon = styled(Icon)`
  position: relative;
  top: 2px;
  margin-right: 16px;
  & svg {
    width: 16px;
    height: 16px;
    stroke: ${secondary};
    & path {
      fill: ${secondary};
      stroke-width: 2;
    }
  }
`;

const Label = styled(Text)`
  color: ${primary};
  font-size: 16px;
  font-weight: 500;
`;

export default () => {
  return (
    <Wrapper onPress={() => console.log('clicked')}>
      <Flex>
        <CustomIcon name='Search' />
        <Label>Where are you going?</Label>
      </Flex>
    </Wrapper>
  );
};
