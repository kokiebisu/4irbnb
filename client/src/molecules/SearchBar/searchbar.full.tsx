import React, { useReducer } from 'react';
import styled, { css } from 'styled-components';
import theme from 'styled-theming';
import { darken } from 'polished';
import { colors } from 'styles';
import { Flex } from 'layout';
import { Box, Button, Icon, Text } from 'atoms';
import { CustomButton } from 'molecules/Button';

type SearchBarProps = {
  onPress: () => void;
};

const hover = theme('mode', {
  light: darken(0.05, colors.white),
});

const seperator = theme('mode', {
  light: darken(0.1, colors.white),
});

const hoverMixin = (number: number) => css`
  &:hover ~ div:nth-child(${number}) {
    width: 1px;
    height: 28px;
    background-color: transparent;
  }
`;

const Wrapper = styled(Box)`
  width: 100%;
  max-width: 850px;
  height: 65px;
  border-radius: 50px;
  display: flex;
  background-color: ${colors.white};
  box-shadow: rgb(0, 0, 0, 0.15) 0px 5px 12px;
`;

const Section = styled(Button)`
  background-color: inherit;
  border-radius: 50px;
  height: 100%;
  width: 100%;
  &:hover {
    background-color: ${hover};
  }
`;

const Seperator = styled(Box)`
  width: 1px;
  height: 28px;
  background-color: ${seperator};
  align-self: center;
`;

const extend = {
  options: css`
    background-color: white;
    display: flex;
    flex: 1 1 0%;
    height: 100%;
    border-radius: inherit;
  `,
  section: {
    first: css`
      flex: 1.5 0 0%;
      order: 1;
      ${hoverMixin(5)}
    `,
    second: css`
      flex: 0.8 0 0%;
      padding: 20px;
      order: 3;
      ${hoverMixin(5)}
      ${hoverMixin(6)}
    `,
    third: css`
      flex: 0.8 0 0%;
      padding: 20px;
      order: 5;
      ${hoverMixin(6)}
      ${hoverMixin(7)}
    `,
    fourth: css`
      flex: 1.15 0 0%;
      order: 7;
      ${hoverMixin(7)}
    `,
  },
  seperator: {
    first: css`
      order: 2;
    `,
    second: css`
      order: 4;
    `,
    third: css`
      order: 6;
    `,
  },
};

export default ({ ...props }: SearchBarProps) => {
  return (
    <Wrapper>
      <Box styles={extend.options}>
        <Section
          styles={extend.section.first}
          onPress={() => console.log('pressed')}>
          sadf
        </Section>
        <Section
          styles={extend.section.second}
          onPress={() => console.log('pressed')}>
          sadf
        </Section>
        <Section
          styles={extend.section.third}
          onPress={() => console.log('pressed')}>
          sadf
        </Section>
        <Section
          styles={extend.section.fourth}
          onPress={() => console.log('pressed')}>
          sdaf
        </Section>
        <Seperator styles={extend.seperator.first} />
        <Seperator styles={extend.seperator.second} />
        <Seperator styles={extend.seperator.third} />
      </Box>
    </Wrapper>
  );
};
