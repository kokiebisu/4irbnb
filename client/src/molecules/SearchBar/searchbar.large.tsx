import React from 'react';
import { css } from 'styled-components';
import theme from 'styled-theming';

// color
import { colorpallete } from '../../styles/colorpallete';

// layout
import { Flex } from '../../layout/flex';

// element
import { Box } from '../../atoms/Box';
import { BaseButton } from '../../atoms/Button';
import { Icon } from '../../atoms/Icon';
import { Text } from '../../atoms/Text';

type SearchBarProps = {
  onPress: () => void;
};

export default ({ ...props }: SearchBarProps) => {
  return (
    <BaseButton styles={extend.wrapper} {...props}>
      <Flex alignCenter>
        <Icon name='search' styles={extend.icon} />
        <Box styles={extend.text}>
          <Text>Where are you going?</Text>
        </Box>
      </Flex>
    </BaseButton>
  );
};

const primary = theme('mode', {
  light: colorpallete.gray__5,
});

const secondary = theme('mode', {
  light: colorpallete.gray__7,
});

const extend = {
  wrapper: css`
    background-color: transparent;
    box-shadow: rgb(0, 0, 0, 0.15) 0px 5px 12px;
    height: 50px;
    width: 100%;
    padding: 0 10px 0 18px;
    border-radius: 24px;
  `,
  icon: css`
    position: relative;
    top: 2px;
    margin-right: 16px;
    & svg {
      width: 16px;
      height: 16px;
      stroke: ${secondary};
      & path {
        fill: ${secondary};
        stroke-width: 2;
      }
    }
  `,
  text: css`
    & p {
      color: ${primary};
      font-size: 16px;
      font-weight: 500;
    }
  `,
};
