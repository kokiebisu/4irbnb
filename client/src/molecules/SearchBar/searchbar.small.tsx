import React from 'react';
import { css } from 'styled-components';
import theme from 'styled-theming';

// layout
import { Flex } from 'layout/flex';

// element
import { Icon } from 'atoms/Icon';
import { Box } from 'atoms/Box';
import { BaseButton } from 'atoms/Button';
import { Text } from 'atoms/Text';

// colors
import { colorpallete } from 'styles/colorpallete';

type SearchBarProps = {
  onPress: () => void;
};

export default ({ ...props }: SearchBarProps) => {
  return (
    <BaseButton styles={extend.wrapper} {...props}>
      <Flex justifyBetween alignCenter>
        <Box styles={extend.text}>
          <Text>Start your search</Text>
        </Box>
        <Box styles={extend.iconwrapper}>
          <Icon name='search' styles={extend.icon} />
        </Box>
      </Flex>
    </BaseButton>
  );
};

const primary = theme('mode', {
  light: [colorpallete.pink__4, colorpallete.pink__5, colorpallete.pink__6],
});

const secondary = theme('mode', {
  light: colorpallete.gray__1,
});

const white = theme('mode', {
  light: colorpallete.white,
});

const extend = {
  wrapper: css`
    background-color: transparent;
    border: 1px solid ${secondary};
    height: 50px;
    max-width: 100%;
    padding: 0 7px 0 18px;
    min-width: 220px;
    border-radius: 24px;
  `,
  iconwrapper: css`
    width: 32px;
    height: 32px;
    background: -webkit-linear-gradient(
      right,
      rgb(230, 30, 77) 0%,
      rgb(227, 28, 95) 50%,
      rgb(215, 4, 102) 100%
    );
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
  `,
  icon: css`
    position: relative;

    & svg {
      width: 12px;
      height: 12px;
      stroke: ${white};
      & path {
        fill: ${white};
        stroke-width: 2;
      }
    }
  `,
  text: css`
    & p {
      color: ${primary};
      font-family: 'Airbnb-Cereal';
      font-size: 14px;
      font-weight: 500;
    }
  `,
};
