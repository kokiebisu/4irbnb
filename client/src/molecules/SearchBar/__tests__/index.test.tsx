import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { SearchBar } from 'molecules/SearchBar';

describe('Searchbar', () => {
  it('renders correctly with small size', () => {
    const testingData = {
      type: 'sm',
      onPress: () => console.log('it is working'),
    };

    const tree = renderer.create(<SearchBar {...testingData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with medium size', () => {
    const testingData = {
      type: 'md',
      onPress: () => console.log('it is working'),
    };

    const tree = renderer.create(<SearchBar {...testingData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with large size', () => {
    const testingData = {
      type: 'lg',
      onPress: () => console.log('it is working'),
    };

    const tree = renderer.create(<SearchBar {...testingData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
