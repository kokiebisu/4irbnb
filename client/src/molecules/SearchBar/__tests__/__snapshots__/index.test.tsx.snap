// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar renders correctly with mobile size 1`] = `
.c3 p {
  font-size: 16px;
  font-weight: 500;
}

.c0 {
  border-radius: 8px;
  font-size: 18px;
  border: none;
  background-color: transparent;
  box-shadow: rgb(0,0,0,0.15) 0px 5px 12px;
  height: 50px;
  width: 100%;
  padding: 0 10px 0 18px;
  border-radius: 24px;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  top: 2px;
  margin-right: 16px;
}

.c2 svg {
  width: 16px;
  height: 16px;
}

.c2 svg path {
  stroke-width: 2;
}

.c4 {
  margin: 0;
  font-family: 'Airbnb-Cereal';
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  className="c0"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      style={Object {}}
    >
      <svg
        viewBox="0 0 56.966 56.966"
      >
        <path
          d="M55.146,51.887L41.588,37.786c3.486-4.144,5.396-9.358,5.396-14.786c0-12.682-10.318-23-23-23s-23,10.318-23,23 s10.318,23,23,23c4.761,0,9.298-1.436,13.177-4.162l13.661,14.208c0.571,0.593,1.339,0.92,2.162,0.92 c0.779,0,1.518-0.297,2.079-0.837C56.255,54.982,56.293,53.08,55.146,51.887z M23.984,6c9.374,0,17,7.626,17,17s-7.626,17-17,17 s-17-7.626-17-17S14.61,6,23.984,6z"
        />
      </svg>
    </div>
    <div
      className="c3"
      style={Object {}}
    >
      <p
        className="c4"
        style={Object {}}
      >
        Where are you going?
      </p>
    </div>
  </div>
</button>
`;

exports[`Searchbar renders correctly with normal size 1`] = `
.c2 p {
  font-family: 'Airbnb-Cereal';
  font-size: 14px;
  font-weight: 500;
}

.c4 {
  width: 32px;
  height: 32px;
  background: -webkit-linear-gradient( right, rgb(230,30,77) 0%, rgb(227,28,95) 50%, rgb(215,4,102) 100% );
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
}

.c0 {
  border-radius: 8px;
  font-size: 18px;
  border: none;
  background-color: transparent;
  border: 1px solid;
  height: 50px;
  max-width: 100%;
  padding: 0 7px 0 18px;
  min-width: 220px;
  border-radius: 24px;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c5 svg {
  width: 12px;
  height: 12px;
}

.c5 svg path {
  stroke-width: 2;
}

.c3 {
  margin: 0;
  font-family: 'Airbnb-Cereal';
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<button
  className="c0"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      style={Object {}}
    >
      <p
        className="c3"
        style={Object {}}
      >
        Start your search
      </p>
    </div>
    <div
      className="c4"
      style={Object {}}
    >
      <div
        className="c5"
        style={Object {}}
      >
        <svg
          viewBox="0 0 56.966 56.966"
        >
          <path
            d="M55.146,51.887L41.588,37.786c3.486-4.144,5.396-9.358,5.396-14.786c0-12.682-10.318-23-23-23s-23,10.318-23,23 s10.318,23,23,23c4.761,0,9.298-1.436,13.177-4.162l13.661,14.208c0.571,0.593,1.339,0.92,2.162,0.92 c0.779,0,1.518-0.297,2.079-0.837C56.255,54.982,56.293,53.08,55.146,51.887z M23.984,6c9.374,0,17,7.626,17,17s-7.626,17-17,17 s-17-7.626-17-17S14.61,6,23.984,6z"
          />
        </svg>
      </div>
    </div>
  </div>
</button>
`;
